### DISCUSSION TITLE
auth/missing-identifier when sign-in through Microsoft IDP

### DISCUSSION THREAD

**Discussion by ihor-panasiuk95** [firebase/firebase-js-sdk#7122] [Q&A]
Last few weeks our customers complain about impossibility to sign-in through Microsoft IDP. By digging deeper we found out that Firebase throws **auth/missing-identifier** error after signing in with popup. 
Google sign-in with popup works well. 
In browser devtools we clearly see that the endpoint https://identitytoolkit.googleapis.com/v1/accounts:createAuthUri returns 400 status code upon sing-in with Microsoft.
This is the request payload:
![image](https://user-images.githubusercontent.com/69238797/224973531-de25c3fe-db6a-44c6-a8aa-92034c86f265.png)

What that could be and how we can debug/fix this issue?
Thanks in advance.

Our stack:
Angular 15
Firebase 9
@angular/fire 7.5

**Comment by ihor-panasiuk95** [comment]
For whom it may concern, I found the pattern.
This error is thrown during login through Microsoft when I use a third party email (in my case, gmail) that MS account created with. For such case I have to get **auth/account-exists-with-different-credential** thrown instead of this error but instead I get **auth/missing-identifier**. When I login with MS account that's created without any third-party email it works fine and does not throw any error, just signs me in.
The same flow through Google provider works fine.

If I use firebase/compat Microsoft login also works fine and for the email I described above it throws the expected **auth/account-exists-with-different-credential** error.
I'm pretty sure this is the issue of the modular Firebase@9.
Will create the repo with reproduction steps and will raise the bug.

**Comment by ihor-panasiuk95** [comment]
This is the expected error I should get:
![image](https://user-images.githubusercontent.com/69238797/227516861-24ecde6e-1934-4ed6-9246-d09f2b890056.png)
But instead I get:
![image](https://user-images.githubusercontent.com/69238797/227516994-0aac9aac-0eff-43ee-aae4-e8adb8c10725.png)

**Comment by baughmann** [comment]
This is a problem still. Was able to reproduce verbatim.

**Comment by IrinaGaly** [comment]
Also having same problem

**Comment by devops-naya** [comment]
Facing same issue when I am integrating firebase with workOS

**Comment by baughmann** [comment]
For anyone coming across this, it's best to just roll your own integration. If I remember I will post a minimal version of how I did it with Microsoft O365. However, there are some good videos from Firebase about it and there's a good section of the docs regarding it as well.

**Comment by user9747** [comment]
@baughmann can you point towards the firebase docs.

**Reply by baughmann** [reply]
Sorry:
 https://firebase.google.com/docs/auth/web/custom-auth

And

https://firebase.google.com/docs/auth/admin/create-custom-tokens
